### PR TITLE
Added test and coverage report in CI

### LINKED ISSUE #28: Add test and coverage report in CI
Make the tests run when a PR is created and also calculate the coverage for the same so that we can get a report of how much of the code is covered by tests,

### PR DESCRIPTION
**Pull request by Salonichoudhary07**
Fixes #28 

## Summary

Runs tests in CI and gives the coverage report.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,3 +13,32 @@ jobs:
...
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r app/requirements.txt
+          pip install pytest pytest-cov
```
Comment (dalmia) [inline .github/workflows/ci.yml:33]: pytest is already in the requirements right?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,3 +13,32 @@ jobs:
...
       - uses: actions/setup-python@v2
       - uses: pre-commit/action@v2.0.3
+
+  test:
+    name: Test cases
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # Install Python.
```
Comment (dalmia) [inline .github/workflows/ci.yml:23]:
```suggestion
      # Install Python
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -13,3 +13,32 @@ jobs:
...
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r app/requirements.txt
+          pip install pytest pytest-cov
+
+      - name: Run Test Cases
+        env:
+          MONGO_AUTH_CREDENTIALS: "mongodb+srv://quiz:p$#p4h7y_Z44R-n@cluster0.uocfg.mongodb.net/quiz?retryWrites=true&w=majority"
```
Comment (dalmia) [inline .github/workflows/ci.yml:37]: why do we need this if we are running mongo mock?

File: app/requirements.txt [hunk 4]
```diff
@@ -8,3 +8,4 @@ python-dotenv==0.20.0
 mongoengine==0.24.1
 pytest==7.1.2
 mongomock==4.0.0
+requests==2.27.1
```
Comment (dalmia) [inline app/requirements.txt:11]: why do we need this?

**Review by dalmia — CHANGES_REQUESTED** [review]

File: app/requirements.txt (hunk 4 above)
Comment (Salonichoudhary07) [reply app/requirements.txt:11]: I had installed it locally but had forgotten to add this.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Salonichoudhary07) [reply .github/workflows/ci.yml:33]: yes

File: .github/workflows/ci.yml (hunk 3 above)
Comment (Salonichoudhary07) [reply .github/workflows/ci.yml:37]: We will need this or else the post and get request will not be accepted, the mock needs to know which DB to mock.

**Pull request closed without merging**